classes/Uploader.php

```diff
@@ -63,6 +63,11 @@ public function getFilePath($file_name = null)
     {
         if (!isset($file_name)) {
             return tempnam($this->getSavePath(), $this->getUniqueFileName());
+        } else {
```

Comment:
I just say that instead of:

```php
if (!isset($file_name)) {
    return tempnam($this->getSavePath(), $this->getUniqueFileName());
} else {
    $path_info = pathinfo($file_name);
    if (isset($path_info['extension'])) {
        $file_name = $path_info['filename'].'.'.Tools::strtolower($path_info['extension']);
    }
}

return $this->getSavePath().$file_name;
```

You should do:

```php
if (!isset($file_name)) {
    return tempnam($this->getSavePath(), $this->getUniqueFileName());
}

$path_info = pathinfo($file_name);
if (isset($path_info['extension'])) {
    $file_name = $path_info['filename'].'.'.Tools::strtolower($path_info['extension']);
}

return $this->getSavePath().$file_name;
```